source/mir/ndslice/slice.d

```diff
@@ -2544,6 +2544,51 @@ public:
         assert(imm == x);
         assert(mut == x);
     }
+    
+    size_t[N] numIndex(size_t n) @safe scope const
+    {
+        assert(n < elementCount, "numIndex: n must be less than elementCount");
+
+        size_t[N] output;
+        static if (N == 1) {
+            output[N - 1] = n;
+        } else {
+            output[0 .. (N - 1)] = _lengths[1 .. N];
+            static if (N > 2) {
+                foreach_reverse (i; Iota!(N - 2)) {
+                    output[i] *= output[i + 1];
+                }
+            }
+            size_t mul;
+            foreach (i; Iota!(N - 1)) {
+                mul = n / output[i];
+                n -= mul * output[i];
+                output[i] = mul;
+            }
+            output[N - 1] = n;
+        }
+        
+        return output;
+    }
+
+    size_t valStride(size_t n) @safe scope const
+    {
+        assert(n < elementCount, "valStride: n must be less than elementCount");
+
+        static if (kind == Contiguous || (kind == Canonical && N == 1)) {
+            return n;
+        } else static if (kind == Universal && N == 1) {
+            return n * _strides[0];
+        } else {
+            return indexStride(numIndex(n));
+        }
+    }
+
+    ///
+    auto ref at(size_t n) scope return @trusted
```

Comment:
Also needs more docs and more verbose name

Reply:
Happy to add more docs later today. Any suggestion on name? Maybe `access` or `accessFlat`? I am flexible.

Along the same lines, should gmean/hmean be renamed geometricMean/harmonicMean? Might need deprecation given that there have been releases...

Reply:
accessFlag looks good. Existing names in stat are good as well.